### ISSUE TITLE
Failed to install  hclust2 on Mac OS

### ISSUE THREAD

**Issue by wn835166087** [biobakery/homebrew-biobakery#32]
hi,
I was trying to install hclust2 on my mac metaphlan-env. But I failed to install the hclust2.
Here are the logs:
(metaphlan-env) vcv075026:metaphlan a123$ brew install hclust2
==> Installing hclust2 from biobakery/biobakery
==> Installing dependencies for biobakery/biobakery/hclust2: bzip2
==> Installing biobakery/biobakery/hclust2 dependency: bzip2
==> Downloading https://homebrew.bintray.com/bottles/bzip2-1.0.6_1.mojave.bottle
######################################################################## 100.0%
==> Pouring bzip2-1.0.6_1.mojave.bottle.tar.gz
==> Caveats
bzip2 is keg-only, which means it was not symlinked into /usr/local,
because macOS already provides this software and installing another version in
parallel can cause all kinds of trouble.

If you need to have bzip2 first in your PATH run:
  echo 'export PATH="/usr/local/opt/bzip2/bin:$PATH"' >> ~/.bash_profile

For compilers to find bzip2 you may need to set:
  export LDFLAGS="-L/usr/local/opt/bzip2/lib"
  export CPPFLAGS="-I/usr/local/opt/bzip2/include"

==> Summary
🍺  /usr/local/Cellar/bzip2/1.0.6_1: 26 files, 400.4KB
==> Installing biobakery/biobakery/hclust2
==> Downloading https://bitbucket.org/nsegata/hclust2/get/3d589ab2cb68.tar.gz
######################################################################## 100.0%
==> Downloading https://pypi.python.org/packages/source/n/numpy/numpy-1.7.1.tar.
==> Downloading from https://files.pythonhosted.org/packages/84/fb/5e9dfeeb5d890
######################################################################## 100.0%
==> python2 -c import setuptools... --no-user-cfg install --prefix=/usr/local/Ce
Last 15 lines from /Users/a123/Library/Logs/Homebrew/hclust2/01.python2:
2018-11-26 08:35:58 -0800

python2
-c
import setuptools, tokenize
__file__ = 'setup.py'
exec(compile(getattr(tokenize, 'open', open)(__file__).read()
  .replace('\r\n', '\n'), __file__, 'exec'))
--no-user-cfg
install
--prefix=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec
--install-scripts=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec/bin
--single-version-externally-managed
--record=installed.txt

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/biobakery/homebrew-biobakery/issues

Error: You have MacPorts or Fink installed:
  /opt/local/bin/port

This can cause trouble. You don't have to uninstall them, but you may want to
temporarily move them out of the way, e.g.

  sudo mv /opt/local ~/macports
(metaphlan-env) vcv075026:metaphlan a123$ sudo mv /opt/local ~/macports
Password:
(metaphlan-env) vcv075026:metaphlan a123$ brew install hclust2
==> Installing hclust2 from biobakery/biobakery
==> Downloading https://bitbucket.org/nsegata/hclust2/get/3d589ab2cb68.tar.gz
Already downloaded: /Users/a123/Library/Caches/Homebrew/downloads/fbb64c454c95e647ae50b5fc7db4af135f7a63e45a2ee555af40d5de5abd4797--nsegata-hclust2-3d589ab2cb68.tar.gz
==> Downloading https://pypi.python.org/packages/source/n/numpy/numpy-1.7.1.tar.
Already downloaded: /Users/a123/Library/Caches/Homebrew/downloads/32246c557b05158cfbb1b88f4aa555c35a721709caa7c670e99fe38e923a59fc--numpy-1.7.1.tar.gz
==> python2 -c import setuptools... --no-user-cfg install --prefix=/usr/local/Ce
Last 15 lines from /Users/a123/Library/Logs/Homebrew/hclust2/01.python2:
2018-11-26 08:41:30 -0800

python2
-c
import setuptools, tokenize
__file__ = 'setup.py'
exec(compile(getattr(tokenize, 'open', open)(__file__).read()
  .replace('\r\n', '\n'), __file__, 'exec'))
--no-user-cfg
install
--prefix=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec
--install-scripts=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec/bin
--single-version-externally-managed
--record=installed.txt

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/biobakery/homebrew-biobakery/issues

(metaphlan-env) vcv075026:metaphlan a123$ brew install hclust2 --without-numpy
==> Installing hclust2 from biobakery/biobakery
==> Downloading https://bitbucket.org/nsegata/hclust2/get/3d589ab2cb68.tar.gz
Already downloaded: /Users/a123/Library/Caches/Homebrew/downloads/fbb64c454c95e647ae50b5fc7db4af135f7a63e45a2ee555af40d5de5abd4797--nsegata-hclust2-3d589ab2cb68.tar.gz
==> Downloading https://pypi.python.org/packages/source/s/scipy/scipy-0.12.0.tar.gz
==> Downloading from https://files.pythonhosted.org/packages/51/1c/d53b2c99f8dd94307ddc92772c7a696b045384c40e039240c24c3b2bae28/scipy-0.12.0.tar.gz
######################################################################## 100.0%
==> python2 -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec --install-scripts=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec/bin --single-version-ex
Last 15 lines from /Users/a123/Library/Logs/Homebrew/hclust2/01.python2:
2018-11-26 08:44:24 -0800

python2
-c
import setuptools, tokenize
__file__ = 'setup.py'
exec(compile(getattr(tokenize, 'open', open)(__file__).read()
  .replace('\r\n', '\n'), __file__, 'exec'))
--no-user-cfg
install
--prefix=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec
--install-scripts=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec/bin
--single-version-externally-managed
--record=installed.txt

If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/biobakery/homebrew-biobakery/issues

(metaphlan-env) vcv075026:metaphlan a123$ brew install python
Error: python 3.7.0 is already installed
To upgrade to 3.7.1, run `brew upgrade python`
(metaphlan-env) vcv075026:metaphlan a123$ brew install hclust2 --without-numpy --verbose
==> Installing hclust2 from biobakery/biobakery
/usr/bin/sandbox-exec -f /private/tmp/homebrew20181126-40698-1bvuyn.sb nice /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby -W0 -I /usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/ruby-macho-2.1.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/rubocop-rspec-1.30.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/rubocop-0.60.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/unicode-display_width-1.4.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/ruby-progressbar-1.10.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/rainbow-3.0.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/powerpack-0.1.2/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/plist-3.4.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/parser-2.5.3.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/parallel-1.12.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/jaro_winkler-1.5.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/extensions/universal-darwin-18/2.3.0/jaro_winkler-1.5.1:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/backports-3.11.4/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/ast-2.4.0/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/activesupport-5.2.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/tzinfo-1.2.5/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/thread_safe-0.3.6/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/minitest-5.11.3/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/i18n-1.1.1/lib:/usr/local/Homebrew/Library/Homebrew/vendor/bundle-standalone/bundler/../ruby/2.3.0/gems/concurrent-ruby-1.0.5/lib:/Library/Ruby/Gems/2.3.0/gems/did_you_mean-1.0.0/lib:/Library/Ruby/Site/2.3.0:/Library/Ruby/Site/2.3.0/x86_64-darwin18:/Library/Ruby/Site/2.3.0/universal-darwin18:/Library/Ruby/Site:/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0:/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/x86_64-darwin18:/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby/2.3.0/universal-darwin18:/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/vendor_ruby:/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0:/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/x86_64-darwin18:/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib/ruby/2.3.0/universal-darwin18:/usr/local/Homebrew/Library/Homebrew -- /usr/local/Homebrew/Library/Homebrew/build.rb /usr/local/Homebrew/Library/Taps/biobakery/homebrew-biobakery/hclust2.rb --verbose --without-numpy
==> Downloading https://bitbucket.org/nsegata/hclust2/get/3d589ab2cb68.tar.gz
Already downloaded: /Users/a123/Library/Caches/Homebrew/downloads/fbb64c454c95e647ae50b5fc7db4af135f7a63e45a2ee555af40d5de5abd4797--nsegata-hclust2-3d589ab2cb68.tar.gz
==> Verifying fbb64c454c95e647ae50b5fc7db4af135f7a63e45a2ee555af40d5de5abd4797--nsegata-hclust2-3d589ab2cb68.tar.gz checksum
tar xf /Users/a123/Library/Caches/Homebrew/downloads/fbb64c454c95e647ae50b5fc7db4af135f7a63e45a2ee555af40d5de5abd4797--nsegata-hclust2-3d589ab2cb68.tar.gz -C /private/tmp/d20181126-40700-99cfwo
cp -pR /private/tmp/d20181126-40700-99cfwo/nsegata-hclust2-3d589ab2cb68/. /private/tmp/hclust2-20181126-40700-1nfmacm/nsegata-hclust2-3d589ab2cb68
chmod -Rf +w /private/tmp/d20181126-40700-99cfwo
==> Downloading https://pypi.python.org/packages/source/s/scipy/scipy-0.12.0.tar.gz
Already downloaded: /Users/a123/Library/Caches/Homebrew/downloads/9957d16b3f57d822d8b05f6f90017aabbdbfffca982b51fa9fbc241800f94282--scipy-0.12.0.tar.gz
==> Verifying 9957d16b3f57d822d8b05f6f90017aabbdbfffca982b51fa9fbc241800f94282--scipy-0.12.0.tar.gz checksum
tar xf /Users/a123/Library/Caches/Homebrew/downloads/9957d16b3f57d822d8b05f6f90017aabbdbfffca982b51fa9fbc241800f94282--scipy-0.12.0.tar.gz -C /private/tmp/d20181126-40700-1ps5ic0
cp -pR /private/tmp/d20181126-40700-1ps5ic0/scipy-0.12.0/. /private/tmp/hclust2--scipy-20181126-40700-jnet7e/scipy-0.12.0
chmod -Rf +w /private/tmp/d20181126-40700-1ps5ic0
==> python2 -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec --install-scripts=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec/bin --single-version-externally-managed --record=installed.txt
Failed to execute: python2

==> Formula
Tap: biobakery/biobakery
Path: /usr/local/Homebrew/Library/Taps/biobakery/homebrew-biobakery/hclust2.rb
==> Configuration
HOMEBREW_VERSION: 1.8.4-3-g6f0ef7d
ORIGIN: https://github.com/Homebrew/brew
HEAD: 6f0ef7deecaf267277f315a4b70fb7277269728c
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e5def19d72af783c8c5e3e763afce0debb9dc51d
Core tap last commit: 2 hours ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_DEV_CMD_RUN: 1
HOMEBREW_ENABLE_AUTO_UPDATE_MIGRATION: 1
CPU: quad-core 64-bit kabylake
Homebrew Ruby: 2.3.7 => /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby
Clang: 10.0 build 1000
Git: 2.17.2 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.54.0 => /usr/bin/curl
Java: 11
macOS: 10.14.1-x86_64
CLT: N/A
Xcode: 10.1
==> ENV
HOMEBREW_CC: clang
HOMEBREW_CXX: clang++
MAKEFLAGS: -j4
CMAKE_PREFIX_PATH: /usr/local/opt/bzip2:/usr/local
CMAKE_INCLUDE_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/libxml2:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/apache2:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
CMAKE_LIBRARY_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
CMAKE_FRAMEWORK_PATH: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks
PKG_CONFIG_LIBDIR: /usr/lib/pkgconfig:/usr/local/Homebrew/Library/Homebrew/os/mac/pkgconfig/10.14
HOMEBREW_GIT: git
HOMEBREW_SDKROOT: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk
ACLOCAL_PATH: /usr/local/share/aclocal
PATH: /usr/local/Homebrew/Library/Homebrew/shims/mac/super:/usr/local/opt/bzip2/bin:/usr/bin:/bin:/usr/sbin:/sbin

Error: biobakery/biobakery/hclust2 1.0.0-3d589ab did not build
Logs:
     /Users/a123/Library/Logs/Homebrew/hclust2/00.options.out
     /Users/a123/Library/Logs/Homebrew/hclust2/01.python2
If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/biobakery/homebrew-biobakery/issues

here are  /Users/a123/Library/Logs/Homebrew/hclust2/00.options.out
     /Users/a123/Library/Logs/Homebrew/hclust2/01.python2
👇
[归档.zip](https://github.com/biobakery/homebrew-biobakery/files/2616556/default.zip)

**Comment by lancetxiao** [comment]
$ brew install hclust2
Updating Homebrew...
==> Installing hclust2 from biobakery/biobakery
==> Downloading https://bitbucket.org/nsegata/hclust2/get/3d589ab2cb68.tar.gz
######################################################################## 100.0%
==> Downloading https://pypi.python.org/packages/source/n/numpy/numpy-1.7.1.tar.gz
==> Downloading from https://files.pythonhosted.org/packages/84/fb/5e9dfeeb5d8909d659e6892c97c9aa66d3798fad50e1d3d66b3c614a9c35/numpy-1.7.1.tar.gz
######################################################################## 100.0%
==> python2 -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec --install-scripts=/usr/local/Cellar/hclust2/1.0.0-
Last 15 lines from /Users/yuxiao/Library/Logs/Homebrew/hclust2/01.python2:
2019-01-19 22:11:53 +0800

python2
-c
import setuptools, tokenize
__file__ = 'setup.py'
exec(compile(getattr(tokenize, 'open', open)(__file__).read()
  .replace('\r\n', '\n'), __file__, 'exec'))
--no-user-cfg
install
--prefix=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec
--install-scripts=/usr/local/Cellar/hclust2/1.0.0-3d589ab/libexec/bin
--single-version-externally-managed
--record=installed.txt


If reporting this issue please do so at (not Homebrew/brew or Homebrew/core):
https://github.com/biobakery/homebrew-biobakery/issues

These open issues may also help:
Failed to install  hclust2 on Mac OS https://github.com/biobakery/homebrew-biobakery/issues/32

**Comment by sytjon** [comment]
@lancetxiao I had the exact same issue. 
The way I solved it was by going to the place where the downloads from homebrew are stored. In my case this was:
/Users/<username>/Library/Caches/Homebrew/downloads/ 
I just opened the zip file manually: 
"fbb64c454c95e647ae50b5fc7db4af135f7a63e45a2ee555af40d5de5abd4797--nsegata-hclust2-3d589ab2cb68.tar" 

to get

a folder: nsegata-hclust2-3d589ab2cb68

In this folder is the hclust2.py file etc. So, I just edited my bashrc file to add this folder to my path.

in terminal: open ~/.bashrc
add this line in txt file:
export PATH=/Users/<username>/Library/Caches/Homebrew/downloads/nsegata-hclust2-3d589ab2cb68/:$PATH 

in terminal: source ~/.bashrc
